src/components/shared/SearchCard/SearchCardAction/DownloadNowModal.js

```diff
@@ -28,17 +28,26 @@ export const DownloadNowModal = ({
   const [formValues, setFormValues] = useState(null)
 
   const submit = async () => {
-    const { emailAddress, receiveUpdates } = formValues
-
-    if (receiveUpdates) {
-      const subscribeEmailResponse = await subscribeEmail(emailAddress)
+    if (formValues.email_ccdl_ok) {
+      const subscribeEmailResponse = await subscribeEmail(
+        formValues.email_address
+      )
       if (subscribeEmailResponse.status !== 'error') {
         gtag.trackEmailSubscription(DownloadNowModal)
       }
     }
 
+    const downloadOptions = {
+      aggregate_by: formValues.aggregate_by,
+      data: { [accessionCode]: ['ALL'] },
+      scale_by: formValues.scale_by,
+      quantile_normalize: formValues.quantile_normalize,
+      email_address: formValues.email_address,
+      email_ccdl_ok: formValues.email_ccdl_ok
+    }
```

Comment:
```suggestion
    const downloadOptions = {
      data: { [accessionCode]: ['ALL'] },
      ...formValues
    }
```

Cant we do something like this?